configure

```diff
@@ -1,387 +1,241 @@
-                                --log-level=trace, and
-                                --with-sanitizers=address,undefined
-
-  Influential Environment Variables (only on first invocation):
-    CXX                         C++ compiler command
-    CXXFLAGS                    C++ compiler flags (overrides defaults)
-    LDFLAGS                     Additional linker flags
-    CMAKE_GENERATOR     Selects a custom generator
+Command="$0 $*"
+CommandDirname=`dirname $0`
+SourceDir=`cd $CommandDirname && pwd`
 
-  iOS Build Options (should be used with XCode generator):
-    --sysroot=DIR               set system root for Clang
-                                  - iphoneos: for iOS device
-                                  - iphonesimulator: for iOS simulator
-    --ios-min-ver=VERSION       set the ios deployment target version
+usage="\
+Usage:
+
+  $0 [--<variable>=<value>...]
+
+General CMake options:
+
+  --cmake=PATH              set a custom path to the CMake binary
+  --build-dir=PATH          set build directory [build]
+  --build-type=STRING       set build type of single-configuration generators
+  --generator=STRING        set CMake generator (see cmake --help)
+  --cxx-flags=STRING        set CMAKE_CXX_FLAGS when running CMake
+  --prefix=PATH             set installation directory
+
+Debugging options:
+
+    --log-level=STRING      build with debugging output, possible values:
+                              - ERROR
+                              - WARNING
+                              - INFO
+                              - DEBUG
+                              - TRACE
+    --sanitizers=STRING     build with this list of sanitizers enabled
+
+Convenience options:
+
+  --dev-mode                shortcut for passing:
+                              --build-type=Debug
+                              --log-level=TRACE
+                              --sanitizers=address,undefined
+                              --disable-examples
+                              --disable-tools
+                              --enable-utility-targets
+                              --enable-runtime-checks
+
+Flags (use --enable-<name> to activate and --disable-<name> to deactivate):
+
+  shared-libs               build shared library targets [ON]
+  export-compile-commands   write JSON compile commands database [ON]
+  prefer-pthread-flag       prefer -pthread flag if available  [ON]
+  curl-examples             build examples with libcurl [OFF]
+  protobuf-examples         build examples with Google Protobuf [OFF]
+  qt5-examples              build examples with the Qt5 framework [OFF]
+  runtime-checks            build CAF with extra runtime assertions [OFF]
+  utility-targets           include targets like consistency-check [OFF]
+  actor-profiler            enable experimental proiler API [OFF]
+  examples                  build small programs showcasing CAF features [ON]
+  io-module                 build networking I/O module [ON]
```

Comment:
What about the openssl module?

Reply:
Good question. Looks like I've missed it.